zrml/prediction-markets/src/benchmarks.rs

```diff
@@ -215,18 +198,72 @@ fn setup_resolve_common_scalar<T: Config>(
     Ok((caller, market_id))
 }
 
-// Setup a disputed scalar market and create accounts with outcome assets.
-fn setup_resolve_common_scalar_after_dispute<T: Config>(
-    acc_total: u32,
-    acc_asset: u32,
+fn setup_reported_categorical_market_with_pool<T: Config>(
+    categories: u32,
 ) -> Result<(T::AccountId, MarketIdOf<T>), &'static str> {
-    let (caller, market_id) = setup_resolve_common_scalar::<T>(acc_total, acc_asset)?;
-    Pallet::<T>::dispute(
+    let (caller, market_id) = create_market_common::<T>(
+        MarketCreation::Permissionless,
+        MarketType::Categorical(categories.saturated_into()),
+        ScoringRule::CPMM,
+        None,
+    )?;
+
+    let max_swap_fee: BalanceOf<T> = MaxSwapFee::get().saturated_into();
+    let min_liquidity: BalanceOf<T> = MinLiquidity::get().saturated_into();
+    let _ = Call::<T>::buy_complete_set { market_id, amount: min_liquidity }
+        .dispatch_bypass_filter(RawOrigin::Signed(caller.clone()).into())?;
```

Comment:
The `let _ =` pattern should be removed here. You're using the question mark operator, so you're already using the result.

Reply:
I guess this is debatable since the call doesn't return a `DispatchResultWithPostInfo`, so consider this a nitpick.